File: dataStream.java
// Very similar to zeroFilledSubarray.java & .py
// Approach 1- Count the no. of consecutive numbers at each index
import java.util.*;
class Solution {
    int value;
    int k;
    int consecSubarray=0;
    
    public Solution(int value, int k) {
        this.value=value;
        this.k=k;
    }
    
    public boolean consec(int num) {
        if (num==value){
            consecSubarray++;
        } else {
            consecSubarray=0;
        }
        return consecSubarray>=k;
    }
}

class dataStream{
    public static void main(String[] args) {
        Solution s= new Solution(4,3);
        System.out.println(s.consec(4));
        System.out.println(s.consec(4));
        System.out.println(s.consec(4));
        System.out.println(s.consec(3));
    }
}

File: dataStream.py
# Very similar to zeroFilledSubarray.java & .py
# Approach 1- Count the no. of consecutive numbers at each index
class DataStream:
    def __init__(self, value, k):
        self.value=value
        self.k=k
        self.count=0

    def consec(self, num: int) -> bool:
        if num==self.value:
            self.count+=1
        else:
            self.count=0
        
        return self.count>=self.k


s=DataStream(4, 3)
print(s.consec(4))
print(s.consec(4))
print(s.consec(4))
print(s.consec(3))
